pennylane/labs/resource_estimation/ops/op_math/symbolic.py

```diff
@@ -49,29 +196,245 @@ def _resource_decomp(
+    ... )
+    >>> controlled_qft.resources(**controlled_qft.resource_params)
+    defaultdict(<class 'int'>, {C(Hadamard,3,0,2): 3, C(SWAP,3,0,2): 1, C(ControlledPhaseShift,3,0,2): 3})
+
+    Alternatively, we can call the resources method on from the class:
+
+    >>> re.ResourceControlled.resources(
+    ...     base_class = re.ResourceQFT,
+    ...     base_params = {"num_wires": 3},
+    ...     num_ctrl_wires = 3,
+    ...     num_ctrl_values = 0,
+    ...     num_work_wires = 2,
+    ... )
+    defaultdict(<class 'int'>, {C(Hadamard,3,0,2): 3, C(SWAP,3,0,2): 1, C(ControlledPhaseShift,3,0,2): 3})
+
+    .. details::
+        :title: Usage Details
+
+        We can configure the resources for the controlled of a base operation by modifying
+        its :code:`.controlled_resource_decomp(num_ctrl_wires, num_ctrl_values, num_work_wires,
+        **resource_params)` method. Consider for example this custom PauliZ class, where the
+        controlled resources are not defined (this is the default for a general :class:`~.ResourceOperator`).
+
+
+        .. code-block:: python
+
+            class CustomZ(re.ResourceZ):
+
+                @classmethod
+                def controlled_resource_decomp(cls, num_ctrl_wires, num_ctrl_values, num_work_wires):
+                    raise re.ResourcesNotDefined
+
+        When this method is not defined, the controlled resources are computed by taking the
+        controlled of each operation in the resources of the base operation.
+
+        >>> CustomZ.resources()
+        {S: 2}
+        >>> re.ResourceControlled.resources(CustomZ, {}, num_ctrl_wires=1, num_ctrl_values=0, num_work_wires=0)
+        defaultdict(<class 'int'>, {C(S,2,0,3): 2})
+
+        We can update the controlled resources with the observation that the PauliZ gate when controlled
+        on a single wire is equivalent to :math:`\hat{CZ} &= \hat{H} \cdot \hat{CNOT} \cdot \hat{H}`.
+        so we can modify the base operation:
+
+        .. code-block:: python
+
+            class CustomZ(re.ResourceZ):
+
+                @classmethod
+                def controlled_resource_decomp(cls, num_ctrl_wires, num_ctrl_values, num_work_wires):
+                    if num_ctrl_wires == 1 and num_ctrl_values == 0:
+                        return {
+                            re.ResourceHadamard.resource_rep(): 2,
+                            re.ResourceCNOT.resource_rep(): 1,
+                        }
+                    raise re.ResourcesNotDefined
+
+        >>> re.ResourceControlled.resources(CustomZ, {}, num_ctrl_wires=1, num_ctrl_values=0, num_work_wires=0)
+        {Hadamard: 2, CNOT: 1}
+
+    """
 
     @classmethod
     def _resource_decomp(
         cls, base_class, base_params, num_ctrl_wires, num_ctrl_values, num_work_wires, **kwargs
     ) -> Dict[re.CompressedResourceOp, int]:
+        r"""Returns a dictionary representing the resources of the operator. The
```

Comment:
Seems like a copy of the main docstring, please check if this is intentional.

Reply:
Yes this is intentional. The resources section explains how this function works. Because this function is private, and the information is important to be visible at the top level, I added it to the class docstring as well.